complex/src/lib.rs

```diff
@@ -970,6 +1054,95 @@ impl<T> FromStr for Complex<T> where
     }
 }
 
+impl<T: Num + Clone + PartialOrd> Num for Complex<T> {
+    type FromStrRadixErr = ParseComplexError<T::FromStrRadixErr>;
+
+    /// Parses `a +/- bi`; `ai +/- b`; `a`; or `bi` where `a` and `b` are of type `T`
+    fn from_str_radix(s: &str, radix: u32) -> Result<Self, Self::FromStrRadixErr>
```

Comment:
This is identical to the `from_str` implementation, apart from nested calls to `from-str_radix` instead, right?  Let's see if we can use a common helper function instead, something like:

```rust
fn from_str_any<T, E, F: Fn(&str) -> Result<T, E>>(s: &str, from: F) -> Result<Complex<T>, E> { ... }
```

- `from_str` calls `from_str_any(s, T::from_str)`
- `from_str_radix` calls `from_str_any(s, |s| T::from_str_radix(s, radix))`